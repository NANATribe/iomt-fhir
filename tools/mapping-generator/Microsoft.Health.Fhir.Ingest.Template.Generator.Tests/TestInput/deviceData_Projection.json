{
  "time": "2022-07-01T12:02:00Z",
  "device": "TestDeviceId",
  "patient": "TestPatientId",
  "encounter": "TestEncounter",
  "correlation": "TestCorrelation",
  "diastolic": 79,
  "systolic": 119,
  "heartRate": 66,
  "oxygenSaturation": 99
}
