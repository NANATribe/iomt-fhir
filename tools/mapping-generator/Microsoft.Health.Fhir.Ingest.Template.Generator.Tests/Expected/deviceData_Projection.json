[
  {
    "templateType": "CalculatedContent",
    "template": {
      "typeMatchExpression": "$..[?(@diastolic && @systolic)]",
      "deviceIdExpression": "$.device",
      "patientIdExpression": "$.patient",
      "encounterIdExpression": "$.encounter",
      "timestampExpression": "$.time",
      "correlationIdExpression": "$.correlation",
      "values": [
        {
          "valueName": "diastolic",
          "valueExpression": "$.diastolic"
        },
        {
          "valueName": "systolic",
          "valueExpression": "$.systolic"
        }
      ],
      "typeName": "Blood Pressure"
    }
  },
  {
    "templateType": "CalculatedContent",
    "template": {
      "typeMatchExpression": "$..[?(@heartRate)]",
      "deviceIdExpression": "$.device",
      "patientIdExpression": "$.patient",
      "encounterIdExpression": "$.encounter",
      "timestampExpression": "$.time",
      "correlationIdExpression": "$.correlation",
      "values": [
        {
          "valueName": "beatsPerMinute",
          "valueExpression": "$.heartRate"
        }
      ],
      "typeName": "Heart Rate"
    }
  },
  {
    "templateType": "CalculatedContent",
    "template": {
      "typeMatchExpression": "$..[?(@oxygenSaturation)]",
      "deviceIdExpression": "$.device",
      "patientIdExpression": "$.patient",
      "encounterIdExpression": "$.encounter",
      "timestampExpression": "$.time",
      "correlationIdExpression": "$.correlation",
      "values": [
        {
          "valueName": "oxygenPercentage",
          "valueExpression": "$.oxygenSaturation"
        }
      ],
      "typeName": "Oxygen Saturation"
    }
  }
]